server/swap/swap.go

```diff
@@ -2336,23 +2325,41 @@ func (s *Swapper) processMatchAcks(user account.AccountID, msg *msgjson.Message,
 		err = s.authMgr.Auth(user, sigMsg, ack.Sig)
 		if err != nil {
 			log.Warnf("processMatchAcks: 'match' ack for match %v from user %v, "+
-				" failed sig verification: %v", matchInfo.match.ID(), user, err)
+				" failed sig verification: %v", match.ID(), user, err)
 			s.respondError(msg.ID, user, msgjson.SignatureError,
 				fmt.Sprintf("signature validation error: %v", err))
 			return
 		}
 
-		// Store the signature in the DB.
+		// Store the signature in the matchTracker. These must be collected
+		// before the init steps begin and swap contracts are broadcasted.
+		match.mtx.Lock()
+		log.Debugf("processMatchAcks: storing valid 'match' ack signature from %v (maker=%v) "+
+			"for match %v (status %v)", user, matchInfo.isMaker, matchID, match.Status)
+		if matchInfo.isMaker {
+			match.Sigs.MakerMatch = ack.Sig
+		} else {
+			match.Sigs.TakerMatch = ack.Sig
+		}
+		match.mtx.Unlock()
+	}
+
+	// Store the signatures in the DB.
+	for i, matchInfo := range matches {
+		ackSig := acks[i].Sig
+		match := matchInfo.match
+
 		storFn := s.storage.SaveMatchAckSigB
 		if matchInfo.isMaker {
 			storFn = s.storage.SaveMatchAckSigA
 		}
+		matchID := match.ID()
 		mid := db.MarketMatchID{
 			MatchID: matchID,
-			Base:    matchInfo.match.Maker.BaseAsset, // same for taker's redeem as BaseAsset refers to the market
-			Quote:   matchInfo.match.Maker.QuoteAsset,
+			Base:    match.Maker.BaseAsset, // same for taker's redeem as BaseAsset refers to the market
+			Quote:   match.Maker.QuoteAsset,
 		}
-		err = storFn(mid, ack.Sig)
+		err = storFn(mid, ackSig)
```

Comment:
The DB operation comes after al the ack sigs are stored in memory so they can all be registered asap.